applications/plugins/Compliant/python/Compliant/easy.py

```diff
@@ -25,84 +32,110 @@ class Opaque(Structure):
 
 
 def Vec(t):
-    '''a parametrized vector class mirroring that of PythonMultiMapping'''
+    '''a parametrized vector class for state vectors mirroring that of
+    PythonMultiMapping
+    '''
     
     class Vector(Structure):
-        _fields_ = (('outer', c_size_t),
+        _fields_ = (('data', POINTER(t)),
+                    ('outer', c_size_t),
                     ('inner', c_size_t),
-                    ('data', POINTER(t)))
+                    )
 
         def numpy(self):
-            shape = self.outer, self.inner
-            return np.ctypeslib.as_array(self.data, shape)
+            return np.ctypeslib.as_array( as_buffer(self.data, self.outer, self.inner) )
         
     return Vector
-                    
 
-class CompressedStorage(Structure):
-    '''struct mirroring eigen sparse matrices internal storage'''
-    _fields_ = (('values', POINTER(c_double)),
-                ('indices', POINTER(c_int)),
-                ('size', c_size_t),
-                ('allocated_size', c_size_t))
-                 
 
-class SparseMatrix(Structure):
-    '''struct mirroring eigen sparse matrices'''
+def Eigen(t):
+    '''a parametrized vector class for eigen vectors'''
+    
+    class Vector(Structure):
+        _fields_ = (('data', POINTER(t)),
+                    ('size', c_int))
+
+        def numpy(self):
+            return np.ctypeslib.as_array( as_buffer(self.data, self.size) )
+        
+    return Vector
+
+
+
+class ScipyMatrix(Structure):
+    '''all the data needed to alias a matrix in scipy'''
     
-    _fields_ = (('options', c_int), # warning: this is actually a bool (rvalue) + enum (options)
-                ('outer_size', c_int),
-                ('inner_size', c_int),
+    _fields_ = (('rows', c_size_t),
+                ('cols', c_size_t),
                 ('outer_index', POINTER(c_int)),
                 ('inner_nonzero', POINTER(c_int)),
-                ('data', CompressedStorage))
+                ('values', POINTER(c_double)),
+                ('indices', POINTER(c_int)),
+                ('size', c_size_t))
+
+
+dll.eigen_sizeof.restype = c_size_t
+dll.eigen_sizeof.argtypes = ()
 
+def as_buffer(ptr, *size):
```

Comment:
comment:
```'''cast a ctypes pointer as a correctly typed (multidimensional array) of given `size`

this is needed to avoid a nasty memleak with numpy.ctypeslib.as_array
'''```